tools/tests/test_lint.py

```
@@ -67,6 +90,221 @@ def test_clean_tree_passes(tmp_path):
+def test_sideways_deps_reaches_the_charter_and_the_hooks(tmp_path):
+    """A skill named by path from `charter/` does not resolve once installed --
+    the skills live in a plugin cache, not at `skills/` beside the reader."""
+    make_clean_tree(tmp_path)
+    (tmp_path / "charter" / "CHARTER.md").write_text(
+        "The bar lives in skills/example-skill/SKILL.md." + chr(10), encoding="utf-8"
+    )
+    findings = [f for f in lint.run(tmp_path) if "sideways" in f]
+    assert len(findings) == 1
+
+
+def test_the_two_shipped_zone_declarations_agree(tmp_path):
```

Ruff (0.16.1)

[warning] Unused function argument: `tmp_path`

(ARG001)